commit 2fdee73e526751180ffc3dc210253f737af05178
Author: Susan Franklin <sfranklin>
Date:   Mon Feb 18 21:16:28 2008 +0000

    Bug 210670 - [prov] [ui] odd progress placement when adding update site

1	1	bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProfilesView.java
1	1	bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/RepositoriesView.java
1	1	bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/dialogs/AddProfileDialog.java
1	1	bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/RepositoryManipulationDialog.java
23	2	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/ProvisioningOperationRunner.java
1	1	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/AddRepositoryDialog.java
0	4	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/operations/RepositoryOperation.java
0	7	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/operations/UndoableProvisioningOperation.java
diff --git a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProfilesView.java b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProfilesView.java
index a44127c..e6bab42 100644
--- a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProfilesView.java
+++ b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProfilesView.java
@@ -66,7 +66,7 @@ public class ProfilesView extends ProvView {
 				}
 			}
 			ProfileOperation op = new RemoveProfilesOperation(ProvAdminUIMessages.Ops_RemoveProfileOperationLabel, (IProfile[]) profilesOnly.toArray(new IProfile[profilesOnly.size()]));
-			ProvisioningOperationRunner.schedule(op, ProfilesView.this.getShell());
+			ProvisioningOperationRunner.run(op, ProfilesView.this.getShell());
 		}
 	}
 
diff --git a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/RepositoriesView.java b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/RepositoriesView.java
index d882333..76232bc 100644
--- a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/RepositoriesView.java
+++ b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/RepositoriesView.java
@@ -46,7 +46,7 @@ abstract class RepositoriesView extends ProvView {
 		}
 
 		public void run() {
-			ProvisioningOperationRunner.schedule(getRemoveOperation(getSelection().toArray()), getShell());
+			ProvisioningOperationRunner.run(getRemoveOperation(getSelection().toArray()), getShell());
 		}
 	}
 
diff --git a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/dialogs/AddProfileDialog.java b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/dialogs/AddProfileDialog.java
index e4c45bb..808516b 100644
--- a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/dialogs/AddProfileDialog.java
+++ b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/dialogs/AddProfileDialog.java
@@ -77,7 +77,7 @@ public class AddProfileDialog extends StatusDialog {
 		}
 		addedProfileId = profileGroup.getProfileId();
 		Map profileProperties = profileGroup.getProfileProperties();
-		ProvisioningOperationRunner.schedule(new AddProfileOperation(ProvAdminUIMessages.AddProfileDialog_OperationLabel, addedProfileId, profileProperties), getShell());
+		ProvisioningOperationRunner.run(new AddProfileOperation(ProvAdminUIMessages.AddProfileDialog_OperationLabel, addedProfileId, profileProperties), getShell());
 	}
 
 	void verifyComplete() {
diff --git a/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/RepositoryManipulationDialog.java b/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/RepositoryManipulationDialog.java
index 2d1a0d7..3075f77 100644
--- a/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/RepositoryManipulationDialog.java
+++ b/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/RepositoryManipulationDialog.java
@@ -204,7 +204,7 @@ public class RepositoryManipulationDialog extends TrayDialog {
 				}
 				if (urls.size() > 0) {
 					RemoveColocatedRepositoryOperation op = new RemoveColocatedRepositoryOperation(ProvSDKMessages.RepositoryManipulationDialog_RemoveOperationLabel, (URL[]) urls.toArray(new URL[urls.size()]));
-					ProvisioningOperationRunner.schedule(op, getShell());
+					ProvisioningOperationRunner.run(op, getShell());
 				}
 			}
 		});
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/ProvisioningOperationRunner.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/ProvisioningOperationRunner.java
index fb27b86..8da34de 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/ProvisioningOperationRunner.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/ProvisioningOperationRunner.java
@@ -25,8 +25,9 @@ import org.eclipse.ui.progress.WorkbenchJob;
 import org.eclipse.ui.statushandlers.StatusManager;
 
 /**
- * Utility methods for running provisioning operations. All operations are
- * run in jobs, with the operation determining whether the job is run in
+ * Utility methods for running provisioning operations.   Operations can either
+ * be run synchronously or in a job.  When scheduled as a job, the operation
+ * determines whether the job is run in
  * the background or in the UI.
  * 
  * @since 3.4
@@ -34,6 +35,26 @@ import org.eclipse.ui.statushandlers.StatusManager;
 public class ProvisioningOperationRunner {
 
 	/**
+	 * Run the provisioning operation synchronously, adding it to the undo history if it
+	 * supports undo.  Should only be used for operations that run quickly.
+	 * @param op The operation to execute
+	 * @param shell provided by the caller in order to supply UI information for prompting the
+	 *            user if necessary. May be <code>null</code>.
+	 */
+	public static void run(ProvisioningOperation op, Shell shell) {
+		try {
+			if (op instanceof IUndoableOperation) {
+				PlatformUI.getWorkbench().getOperationSupport().getOperationHistory().execute((IUndoableOperation) op, null, ProvUI.getUIInfoAdapter(shell));
+			} else {
+				op.execute(null, ProvUI.getUIInfoAdapter(shell));
+			}
+		} catch (ExecutionException e) {
+			ProvUI.handleException(e.getCause(), NLS.bind(ProvUIMessages.ProvisioningOperationRunner_ErrorExecutingOperation, op.getLabel()), StatusManager.SHOW | StatusManager.LOG);
+		}
+
+	}
+
+	/**
 	 * Schedule a job to execute the supplied ProvisioningOperation, and add it to the
 	 * undo history if it supports undo.
 	 * 
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/AddRepositoryDialog.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/AddRepositoryDialog.java
index c907b94..7e63c67 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/AddRepositoryDialog.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/AddRepositoryDialog.java
@@ -142,7 +142,7 @@ public abstract class AddRepositoryDialog extends StatusDialog {
 
 		URL newURL = makeRepositoryURL(userURL);
 		if (newURL != null) {
-			ProvisioningOperationRunner.schedule(getOperation(newURL), getShell());
+			ProvisioningOperationRunner.run(getOperation(newURL), getShell());
 			return Status.OK_STATUS;
 		}
 		return new Status(IStatus.ERROR, ProvUIActivator.PLUGIN_ID, 0, ProvUIMessages.AddRepositoryDialog_InvalidURL, null);
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/operations/RepositoryOperation.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/operations/RepositoryOperation.java
index 7602b8a..8cce0ad 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/operations/RepositoryOperation.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/operations/RepositoryOperation.java
@@ -43,8 +43,4 @@ abstract class RepositoryOperation extends UndoableProvisioningOperation {
 		return urls;
 	}
 
-	public boolean runInBackground() {
-		return true;
-	}
-
 }
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/operations/UndoableProvisioningOperation.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/operations/UndoableProvisioningOperation.java
index 2267f7a..fad0c57 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/operations/UndoableProvisioningOperation.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/operations/UndoableProvisioningOperation.java
@@ -147,13 +147,6 @@ abstract class UndoableProvisioningOperation extends ProvisioningOperation imple
 	 */
 	public IStatus computeExecutionStatus(IProgressMonitor monitor) {
 		return okStatus();
-	} /*
-	 * (non-Javadoc)
-	 * @see org.eclipse.core.commands.operations.IAdvancedUndoableOperation2#runInBackground()
-	 */
-
-	public boolean runInBackground() {
-		return false;
 	}
 
 	/*
